### PR TITLE
Deprecate jira provider in favor of atlassian.jira provider

### PR DESCRIPTION
**Pull request by eladkal**
Related https://github.com/apache/airflow/issues/25669
**This PR does not change functionality(as requested in the issue) but only names.**

Why I think this name change is better?

1. Jira is Atlassian product and its widely known as such. All atlassian products are under https://www.atlassian.com/

2.  Unlike the previous case of mailing list discussion [Salesforce <-> Tableau](https://lists.apache.org/thread/t9fhhqsdb1x0clg21363s83v27orzk0f) issue https://github.com/apache/airflow/issues/13614 here it makes sense to unite all under 1 provider:

    -  Jira is known to be an Atlassian product. Tableau is not known to be a Salesforce product.
    -  There is no specific site for Jira. There is one for Tableau.
    -  Tableau has its own SDK in it's own domain [tableauserverclient](https://github.com/tableau/server-client-python)  while jira SDK is a module in [atlassian-python-api](https://pypi.org/project/atlassian-python-api/)

3.  We will be able to support more Atlassian integration easily without need to have a new provider for each one of them. This is mentioned as motivation  in https://github.com/apache/airflow/issues/25669

PR info:
1. No system test changes as jira provider doesn't have any
2. No doc changes (except mandatory rst files) as jira provider doesn't have docs.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by eladkal** [comment]
@potiuk could use your help to find the issue with:
```
  #55 17.81 ERROR: Could not find a version that satisfies the requirement apache-airflow-providers-atlassian-jira (from apache-airflow[devel-ci]) (from versions: none)
  #55 17.81 ERROR: No matching distribution found for apache-airflow-providers-atlassian-jira
```

from what I see `devl-ci` providers are taking from `generated/provider_dependencies.json`
so any clue why it's missing?

File: setup.py
Comment (eladkal) [inline setup.py:620]: @potiuk please double check me on the changes to `setup.py`
I had to change `devel_all` as well to make the CI recognize the new provider

Comment (potiuk) [reply setup.py:620]: Yeah. That's right :)

Comment (eladkal) [reply setup.py:620]: Great. Everything should be green now

**Review by potiuk — APPROVED** [review]

**Comment by eladkal** [comment]
@potiuk there is something strange with pre-commit.
It passes locally but in CI it produce weird error - it looks like something is broken?
https://github.com/apache/airflow/runs/8088945322?check_suite_focus=true#step:10:1473

**Comment by potiuk** [comment]
Quote from the output:

```
 If you see that  images got re-generated in CI, please run:

    breeze setup regenerate-command-images
```

**Comment by potiuk** [comment]
If you resolved the conflict manually - this might happen. 

Actually: https://github.com/apache/airflow/pull/26032 is supposed to handle it much better (needs approval).

**Comment by potiuk** [comment]
From: #26032 

> This change also introduces much more selective image generation
which will make the image generation much faster in case you
change any of breeze code. Previously, any change in any breeze
command parameters triggered regeneration of the images.
With this change, only those commands that get changed get
regenerated (and information about which images need to be regenerated
is printed).

**Comment by potiuk** [comment]
We have also the comment in the hash image:

> This file is automatically generated by pre-commit. If you have a conflict with this file
Please do not solve it but run `breeze setup regenerate-command-images`.
This command should fix the conflict and regenerate help images that you have conflict with.

But I understand how this could get wrong so #26032 will make it far more consistent and automated.

**Comment by potiuk** [comment]
Now after #26302 is merged  you need to rebase and regenerate the imges rather than resolve the conflicts one more time :) . Shoudl be the last one :)

**Comment by potiuk** [comment]
BTW: @eladkal - after the #26032 merged the output will be far more readable and you will know more what is going on:

Example after solving some of my conflicts in #25678 

image

**Comment by eladkal** [comment]
@potiuk finally green build :) 
can you please re-review? there were changes since your approval

**Comment by potiuk** [comment]
LGTM. I like that we have the opportunity to test such rename  on such a small provider with low impact. I wonder what other problems we will find when we release (I am sure we will) - but it looks like a good testing ground simillarly to comon.sql and we wil learn if there are other problems connected

**Comment by eladkal** [comment]
yeah extracting time/date related operators from core to common.time (or some other name) is also on my todo list

**Comment by potiuk** [comment]
We can also add "common.taskflow" as described in https://github.com/apache/airflow/issues/26016#issuecomment-1231698030

**Comment by potiuk** [comment]
cc: @uranusjr

**Pull request merged**
